design/specs/record-page.md

```diff
@@ -1,71 +1,137 @@
+## Terminology
 
-## Structure of the record page
-> Not the final layout, but just an outline of what will be included on the record page, so take that into consideration when reading.
-{.is-warning}
+Within this spec...
 
-![image](/assets/design/specs/record-page/179816131-71132b6e-c36d-4265-800a-02f6ae4d42c5.png)
+- **"this record"** refers to the record being displayed on the page
+- **"this table"** refers to the table containing this record
 
-### A.Record-Level Navigation 
+## Parts of the page
 
-The buttons for navigating to different records in the table will be located in the Record-Level Navigation. Users will also be able to search for a specific one using the record selector feature.
+- The page contains the following parts, all of which are described in greater detail below
+    - Title
+    - Direct fields
+    - Widgets
+- The layout of these parts within the page (and of the content within each part) is subject to improvisation during implementation. Because continued adjustment of layout design is relatively low-cost, we are currently prioritizing the implementation of basic functionality over the perfection of aesthetics within this feature.
 
-### B.Record Toolbar 
+## Title
 
-The record toolbar displays the [record summary](#record-summary) for the currently active record. The record toolbar label will also toggle a menu with record-related operations such as duplicate, delete, among others. Some action buttons will be located in the toolbar, similar to the toolbar in tables, however there is no definition of which actions will be provided at this time.
+- One line in large font size. It contains:
+    - A "record" icon. (E.g. something like [this](https://fontawesome.com/icons/memo-pad?s=solid))
+    - The **Record Summary** string
+    - A dropdown icon that opens a menu to perform the following actions on the record:
+        - Delete
+        - *Potentially other actions in the future, such as "Add field", "Duplicate", etc.*
 
-Once the entire flow is specified, this design is likely to change.
+## Direct Fields
 
-### C.Record Fields 
+- This is a section of the page which comes after the title.
+- It contains a form-like area which has many labeled input fields. One field exists for each column in this table.
+- The fields are presented in the same order that they occur in the Table Page.
+- Field labels display the column name along with the icon used within the column header on the Table Page.
+- Edits to the field values are saved on blur, with a similar UX to editing cell values in a table. There is no save button.
+- Similar to the Table Page, a persistent but subtle status message appears at the top of all the fields which reads "Saving" or "All changes saved" when appropriate.
+- FK fields display the record summary for the linked record and allow the user to navigate to its Record Page.
```

Comment:
How do we handle navigating to a record vs updating or deleting a record?

Reply:
Yeah, that's a good question. By _"updating or deleting a record"_, I assuming you mean _"updating the FK cell value, or setting it to NULL"_. This question applies in many other contexts too. In #1467, I have begun building a `LinkedRecordInput` component which handles this behavior. It'll be used for FK cells in the Table Page, for FK fields on the Record Page, for FK inputs within the Record Selector, for setting the default value of an FK column, and for setting the filter condition value when filtering an FK column. It may need to have some slight differences in behavior across those various contexts. For example, maybe we don't need to provide the navigation capability within a filter condition. I'm not sure. 

I think the best path forward here is for me to continue improvising it within the Record Selector implementation and for you to review that work once I feel it's complete. At that point we can have a more productive discussion critiquing the `LinkedRecordInput` design because we'll have something more tangible. Is that approach okay with you, @ghislaineguerin? Or would you like to see more detail added to the specs within this PR?

Reply:
I was also wondering the same thing as @ghislaineguerin. I just want to make sure we handle navigation, updates, and deletes. Improvising the design during implementation sounds like a good plan to me.

Reply:
Update: I'm proposing more detail in #50